compiler/src/iree/compiler/Codegen/Common/EncodingUtils.cpp

```diff
@@ -79,8 +80,36 @@ MaterializeEncodingConversionTarget::MaterializeEncodingConversionTarget(
   });
 }
 
+IREE::Codegen::MaterializeEncodingInfo
+MaterializeEncodingTypeConverter::getEncodingInfo(RankedTensorType type) const {
+  // If the layout is present in the encoding, use it directly. It means that
+  // the layout is already resolved and some information could be dropped during
+  // the lowering. Thus, we prioritize the resolved layout.
+  auto maybeEncodingInfo = getEncodingInfoFromLayouts(type);
+  if (maybeEncodingInfo) {
+    return maybeEncodingInfo.value();
+  }
+  return layoutAttr.getEncodingInfo(type);
+}
+
 RankedTensorType dropEncoding(RankedTensorType type) {
   return RankedTensorType::get(type.getShape(), type.getElementType());
 }
 
+std::optional<IREE::Codegen::MaterializeEncodingInfo>
+getEncodingInfoFromLayouts(RankedTensorType type) {
+  auto encodingAttr = IREE::Encoding::getEncodingAttr(type);
+  if (!encodingAttr) {
+    return std::nullopt;
+  }
+  auto layoutsAttr = encodingAttr.getLayouts();
+  if (!layoutsAttr) {
+    return std::nullopt;
+  }
+  ArrayRef<Attribute> layouts = layoutsAttr.getValue();
+  assert(layouts.size() == 1 && "only single layout is supported");
+  return cast<IREE::Codegen::LayoutAttrInterface>(layouts[0])
```

Comment:
```suggestion
  return cast<IREE::Codegen::LayoutAttrInterface>(layouts.front())
```

Reply:
I like `[0]` better. I think we don't have any convention for this, marking it resolved. Let me know if you have other thought.